src/preamble.js

```diff
@@ -761,23 +761,38 @@ function getBinary() {
 }
 
 function getBinaryPromise() {
-  // If we don't have the binary yet, and have the Fetch api, use that;
-  // in some environments, like Electron's render process, Fetch api may be present, but have a different context than expected, let's only use it on the Web
-  if (!wasmBinary && (ENVIRONMENT_IS_WEB || ENVIRONMENT_IS_WORKER) && typeof fetch === 'function'
+  // If we don't have the binary yet, try to to load it asynchronously.
+  // Fetch has some additional restrictions over XHR, like it can't be used on a file:// url.
+  // (Electron apps are typically loaded from a file)
+  // So use fetch if it is available and the url is not a file, otherwise fall back to XHR
+  if (!wasmBinary && (ENVIRONMENT_IS_WEB || ENVIRONMENT_IS_WORKER)) {
+    if (typeof fetch === 'function'
 #if ENVIRONMENT_MAY_BE_WEBVIEW
-      // Let's not use fetch to get objects over file:// as it's most likely Cordova which doesn't support fetch for file://
+      // Let's not use fetch to get objects over file:// as it's most likely Cordova or Electron which doesn't support fetch for file://
```

Comment:
also please add this link: https://github.com/github/fetch/pull/92#issuecomment-140665932

This seems to be more general than webview, based on that link?